AWSClientRuntime/Tests/Middlewares/Sha256TreeHashMiddlewareTests.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import ClientRuntime
+import AwsCommonRuntimeKit
+@testable import AWSClientRuntime
+
+class Sha256TreeHashMiddlewareTests: XCTestCase {
+    func testTreeHashAllZeroes() {
+        let context = HttpContextBuilder().build()
+        let byteArray: [UInt8] = Array(repeating: 0, count: 5767168)
+        let byteBuffer = ByteBuffer(bytes: byteArray) // 5.5MB buffer
+        let streamInput = StreamInput(body: ByteStream.buffer(byteBuffer))
+        var stack = OperationStack<StreamInput, StreamOutput, StreamError>(id: "TreeHashMiddlewareTestStack")
+        stack.serializeStep.intercept(position: .before, middleware: MockBodyMiddleware())
+        stack.finalizeStep.intercept(position: .after, middleware: Sha256TreeHashMiddleware())
+        _ = stack.handleMiddleware(context: context, input: streamInput, next: MockHandler(callback: { context, input in
+            let linear = input.headers.value(for: "X-Amz-Content-Sha256")
+            XCTAssertEqual(linear, "czz1E0SM5rIK0bxeUOsnwGrvrgwyBxOl3Zn05RvBymA=")
+            let treeHash = input.headers.value(for: "X-Amz-Sha256-Tree-Hash")
+            XCTAssertEqual(treeHash, "x7O1UHyNIXvn/g19LDbAILBPDnqTAz6FwI+MsNCADq0=")
+        }))
```

Comment:
If this callback doesn't get called, then we will pass the test -- One possible solution is that we use an `expectation()` to ensure that this callback gets called.

Reply:
corrected